src/encoder.rs

```diff
@@ -1209,12 +1209,20 @@ pub fn encode_tx_block<T: Pixel, W: Writer>(
       residual,
       &ts.input_tile.planes[p].subregion(area),
       &rec.subregion(area),
-      visible_tx_w,
+      tx_size.width(),
       visible_tx_h,
     );
+    if visible_tx_w < tx_size.width() {
+      for row in residual.chunks_mut(tx_size.width()).take(visible_tx_h) {
+        for a in &mut row[visible_tx_w..] {
+          *a = 0;
+        }
+      }
+    }
   }
-  let visible_area = visible_tx_w * visible_tx_h;
-  for a in residual[visible_area..].iter_mut() {
+  let initialized_area =
+    if visible_tx_w == 0 { 0 } else { tx_size.width() * visible_tx_h };
```

Comment:
I believe "visible_tx_w == 0" case should not reach here, bcz of "if visible_tx_w != 0 && visible_tx_h != 0 { " at line 1207?

Reply:
The converse is true - because of that condition, nothing is initialised in the `visible_tx_w == 0` case.
Specifically, if `visible_tx_h != 0` is also true we need to be cautious not to assume any rows have been initialised.